enhancements/machine-api/converting-machine-api-to-cluster-api.md

```diff
@@ -0,0 +1,756 @@
+- At least one platform has sufficient conversion logic to be useful and allow users on the platform to start migration
+
+Note, we will not require 100% feature parity for promotion to GA.
+Some features supported by Machine API may be supported in conversion only in subsequent releases.
+
+#### Removing a deprecated feature
+
+No features will be deprecated as a part of this enhancement.
+
+### Upgrade / Downgrade Strategy
+
+#### On Upgrade
+
+When upgrading into a release with the new synchronisation operator,
+new Cluster API resources will be created and existing Machine API resources will be annotated.
+
+There should be no effect on the cluster as each of these operations should be a no-op.
+
+#### On Downgrade
+
+On downgrade, the synchronisation of resources will stop.
+To prevent both Machine API and Cluster API resources being reconciled,
+we will make sure that Cluster API controllers in the 4.N-1 release do not reconcile when a mirror resource exists with the same name.
+Independent of the authoritative annotation.
+
+This will mean that on downgrade all Machine management reverts to Machine API controllers.
+
+### Version Skew Strategy
+
+In 4.N-1, all resources that are mirrored into Cluster API will be managed by the Machine API controllers.
+The mirroring relationship will be based on the names of the resources to create a 1:1 mapping between Machines and MachineSets/ControlPlaneMachineSets in Machine API and Cluster API.
+If the Cluster API controller observes a mirror resource, it will not reconcile, independent of the value of the authoritative annotation.
+
+During upgrade, we expect all Machines to be authoritative in Machine API, however, if resources are marked as authoritative in Cluster API,
+there may be a small amount of time[^10] during upgrade where the resources are not reconciled, between the time that the Machine API controllers are upgraded and the Cluster API controllers are upgraded.
+
+[^10]: We must ensure that Cluster API controllers are upgraded _after_ Machine API controllers to ensure that this upgrade is safe.
+The Machine API controllers must observe the annotation and stop reconciling before the Cluster API controllers observe the annotation and act upon the value.
+This can be done by running the Cluster API resources at run-level 31 for the release in which this upgrade is handled.
+
+### Operational Aspects of API Extensions
+
+With the introduction of the new authoritative annotation, `machines.openshift.io/authoritative-api`, this will be validated by a new Machine API webhook.
+The annotation has two defined values and any resource with an invalid value will be rejected as described [above][validating-authority].
+This webhook will run in, and be lifecycled within the new synchronisation controller project.
+
+The new synchronisation controller will operator in the `openshift-machine-api` namespace alongside existing Machine API controllers.
```

Comment:
Do you plan this to be a controller within the already existing manager in the machine-api-operator, or something else?

Reply:
I had intended to start a new repository so that we can keep things clean, and eventually, delete the MAO repo 🤞 

Reusing the same repo means we take on all the old tech debt that exists there

Reply:
Agreed. Makes sense. Thanks